src/main/java/meteordevelopment/meteorclient/systems/modules/render/blockesp/ESPBlockDataScreen.java

```diff
@@ -110,6 +113,20 @@ public void initWidgets() {
             .build()
         );
 
+        // Add state filters setting
+        sgFilters.add(new StringListSetting.Builder()
+            .name("NBT-Data")
+            .description("Filters with states (e.g. 'waterlogged=false', 'facing=north', 'ominous=true'). Only blocks matching ALL filters will be shown.")
+            .defaultValue(new ArrayList<>())
+            .onModuleActivated(stringSetting -> stringSetting.set(blockData.stateFilters))
+            .onChanged(filters -> {
+                blockData.stateFilters.clear();
+                blockData.stateFilters.addAll(filters);
+                onChanged();
+            })
+            .build()
+        );
```

Comment:
imho using a string list for this is super unintuitive, it should be a separate setting type